### ISSUE TITLE
✨ Add support for defineOptions in `require-name-property` rule

### ISSUE THREAD

**Issue by jd-solanki** [vuejs/eslint-plugin-vue#2196]
**What rule do you want to change?**
This doesn't support defineOptions ATM. If we leave defineOptions with empty object

**Does this change cause the rule to produce more or fewer warnings?**
N/A

**How will the change be implemented? (New option, new default behavior, etc.)?**
It should be default behavior

**Please provide some example code that this change will affect:**

```vue
<script lang="ts" setup>
// Below should throw error
defineOptions({})
</script>
```

**What does the rule currently do for this code?**
No error

**What will the rule do after it's changed?**
Should throw error like we do in options API

**Additional context**

N/A

**Comment by jd-solanki** [comment]
> but I think it should be an option

Maybe we can use it as "explicit: true" because name is automatically inferred?
